apps/builder/app/builder/features/project-settings/section-general.tsx

```diff
@@ -129,6 +130,52 @@ export const MetaSection = () => {
           onChange={handleSave("code")}
         />
       </Grid>
+
+      <Separator />
+
+      <CompilerSection />
     </>
   );
 };
+
+const defaultCompilerSettings: CompilerSettings = {
+  atomicStyles: true,
+};
+
+const CompilerSection = () => {
+  const ids = useIds(["atomicStyles"]);
+  const [settings, setSettings] = useState(
+    () => $pages.get()?.compiler ?? defaultCompilerSettings
+  );
+
+  const handleSave = (settings: CompilerSettings) => {
+    serverSyncStore.createTransaction([$pages], (pages) => {
+      if (pages === undefined) {
+        return;
+      }
+      pages.compiler = settings;
+    });
+  };
+
+  return (
+    <Grid gap={2} css={{ mx: theme.spacing[5], px: theme.spacing[5] }}>
+      <Label htmlFor={ids.atomicStyles}>Compiler</Label>
+      <CheckboxAndLabel>
+        <Checkbox
+          checked={settings.atomicStyles ?? true}
+          id={ids.atomicStyles}
+          onCheckedChange={(atomicStyles) => {
+            if (typeof atomicStyles === "boolean") {
```

Comment:
we may want to exclude indeterminate from our component. Caz here I don't know at what point it will appear and block user input.

Reply:
I think it can only be indeterminate if there is no default checked value